Extensions/NavMeshPathfindingBehavior/navmeshpathfindingruntimebehavior.ts

```diff
@@ -0,0 +1,443 @@
+      return this._pathFound;
+    }
+
+    /**
+     * Return true if the object reached its destination.
+     */
+    destinationReached() {
+      return this._reachedEnd;
+    }
+
+    /**
+     * Compute and move on the path to the specified destination.
+     */
+    moveTo(runtimeScene: gdjs.RuntimeScene, x: float, y: float) {
+      //TODO Add a non-blocking padding property to make the path farer from the obstacle if possible?
+      // It may need to have several contour lines with a one cell increment
+      // and do a dichotomy like this:
+      // * find path with obstacleCellPaddingMax (infinity if not found)
+      // * find path with obstacleCellPaddingMin
+      // * path length delta < detourMax ?
+      //   * yes: take PaddingMax path
+      //   * no: do the dichotomy
+      // The contour lines may be needed if there is objects of different size anyway.
+
+      let radiusSqMax = 0;
+      // TODO the center may not be the best thing to use if the object doesn't rotate
+      const centerX = this.owner.getCenterXInScene();
+      const centerY = this.owner.getCenterYInScene();
+      for (const hitBox of this.owner.getHitBoxes()) {
+        for (const vertex of hitBox.vertices) {
+          const deltaX = vertex[0] - centerX;
+          // to have the same unit on x and y
+          const deltaY = (vertex[1] - centerY) * this._manager._isometricRatio;
+          const radiusSq = deltaX * deltaX + deltaY * deltaY;
+          radiusSqMax = Math.max(radiusSq, radiusSqMax);
+        }
+      }
+      // Round to avoid to flicker between 2 NavMesh
+      // because of trigonometry rounding errors.
+      // Round the padding on cellSize to avoid almost identical NavMesh
+      const obstacleCellPadding = Math.max(
+        0,
+        Math.round(
+          (Math.sqrt(radiusSqMax) + this._extraBorder) / this._manager._cellSize
+        )
+      );
+      const navMesh = this._manager.getNavMesh(obstacleCellPadding);
+      this._lastUsedNavMesh = navMesh;
+
+      //TODO The pathfinding could use something like RBush to look for
+      // the the polygons at start and target locations.
+      //TODO If the target is not on the mesh, find the nearest position
+      // maybe the same with the origin to avoid to be stuck.
+      const path = navMesh.findPath(
+        {
+          x: this.owner.getX(),
+          y: this.owner.getY() * this._manager._isometricRatio,
+        },
+        { x: x, y: y * this._manager._isometricRatio }
+      );
+      if (path) {
+        this._pathFound = true;
+        this._path = path.map(({ x, y }) => [x, y]);
+        this._enterSegment(0);
+        return;
+      }
+      // No path found
+      this._pathFound = false;
+    }
+
+    /**
+     * Draw the navigation mesh on a shape painter object for debugging purpose
+     * @param shapePainter
+     */
+    drawNavMesh(shapePainter: gdjs.ShapePainterRuntimeObject) {
+      //TODO should it check that shapePainter is not null?
```

Comment:
You should definitely handle a null object yep!

Reply:
This could be used in the [debug view](https://github.com/4ian/GDevelop/blob/master/GDJS/Runtime/runtimescene.ts#L46)  to the future.

Reply:
Yep I wonder if we could expose a `PIXI.Graphics` object for the debug view, for any extension that wants to draw stuff on it? Rather than using a shape painter object

Reply:
Great idea, furthermore it will be usable in IDE too? My main idea was to be able to get the same debug view in runtime as in the IDE, notwithstanding to add buttons for it.

Reply:
In case of the NavMesh, there can be several meshes depending on the instances size. If it's integrated to the debugger, it may need a UI to select the instances to draw because with 3 meshes or more it will be hard to read.